### PR TITLE
Added Signin and Signout

### PR DESCRIPTION
**Pull request by ChristianCastr0**

### CODE REVIEW

File: server/firebase/firebase.js [hunk 1]
```diff
@@ -23,6 +23,8 @@ const devConfig = {
 const config =
   process.env.NODE_ENV === 'production' ? prodConfig : devConfig;
 
+const provider = new firebase.auth.GoogleAuthProvider();
```
Comment (nguyenhoa2896) [inline server/firebase/firebase.js]: const provider should be inside class Firebase constructor
should change `const provider` to`this.googleProvider`

File: server/firebase/firebase.js [hunk 2]
```diff
@@ -32,40 +34,32 @@ class Firebase {
     this.database = app.database();
```
Comment (nguyenhoa2896) [inline server/firebase/firebase.js:21]: should add `this.adth = app.auth()`;

**Review by nguyenhoa2896 — CHANGES_REQUESTED** [review]

File: server/firebase/firebase.js [hunk 3]
```diff
@@ -23,6 +23,8 @@ const devConfig = {
 const config =
   process.env.NODE_ENV === 'production' ? prodConfig : devConfig;
 
+const provider = new firebase.auth.GoogleAuthProvider();
+
```
Comment (nguyenhoa2896) [inline server/firebase/firebase.js]: should have `import 'firebase/auth';`

**Review by nguyenhoa2896 — CHANGES_REQUESTED** [review]

**Review by nguyenhoa2896 — APPROVED** [review]

File: server/firebase/firebase.js (hunk 3 above)
Comment (ChristianCastr0) [reply server/firebase/firebase.js]: Fixed

**Pull request merged**